### ISSUE TITLE
Window doesn't display until it's updated after switching workspaces

### ISSUE THREAD

**Issue by nashley** [p-e-w/finalterm#326]
When switching workspaces, the window will not be fully drawn (only showing the top bar and the blinking cursor) until the window is updated (i.e., a command is run and output is displayed, the user scrolls, etc.).

For this reason, F10 (i.e., toggling visibility) will not display the window, but F12 (i.e., toggle dropwdown) does; this is because that the dropdown is animated and forces the whole window to update.
Screenshots can be found [here](http://imgur.com/a/zXVYa).
##

**Comment by fabsh** [comment]
Yeah, I'm seeing the same bug under Xfce. Turning compositing off doesn't fix it either.
